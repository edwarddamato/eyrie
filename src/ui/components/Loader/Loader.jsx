import React from 'react';
import './Loader.scss';

const Loader = () => (
  <section className="loader">
    <div className="loader_block loader_block--1"></div>
    <div className="loader_block loader_block--2"></div>
    <div className="loader_block loader_block--3"></div>
    <div className="loader_block loader_block--4"></div>
  </section>
);

export default Loader;
