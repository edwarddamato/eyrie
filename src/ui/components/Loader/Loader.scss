@import "../../styles/base";

.loader {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.8);
  z-index: 99999;

  &_block {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 2rem;
    height: 2rem;
    background-color: $secondary-color;
    animation: loader-big-spin infinite .7s ease-in;
    border: 1px solid transparent;

    &--1 {
      margin: -1rem 0 0 -1rem;
    }
    &--2 {
      margin: -1rem 0 0 1rem;
    }
    &--3 {
      margin: 1rem 0 0 -1rem;
    }
    &--4 {
      margin: 1rem 0 0 1rem;
    }
  }
}

@keyframes loader-big-spin {
  0% { border-color: transparent; transform: translate(-50%, -50%) rotate(0deg); border-radius: 0; }
  25% { transform: translate(-50%, -50%) rotate(45deg); border-radius: 0; }
  50% { transform: translate(-50%, -50%) rotate(45deg); border-radius: 50%; }
  75% { border-color: $white; transform: translate(-50%, -50%) rotate(90deg); border-radius: 50%; }
  100% { border-color: transparent; transform: translate(-50%, -50%) rotate(90deg); border-radius: 0; }
}