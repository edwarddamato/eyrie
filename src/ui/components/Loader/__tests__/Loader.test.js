import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../';

describe('Loader', () => {
  it('renders correctly', () => {
    const loader = renderer.create(
      <Loader />
    ).toJSON();
    expect(loader).toMatchSnapshot();
  });
});
