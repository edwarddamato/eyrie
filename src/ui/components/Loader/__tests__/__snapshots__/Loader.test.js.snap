// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly 1`] = `
<section
  className="loader"
>
  <div
    className="loader_block loader_block--1"
  />
  <div
    className="loader_block loader_block--2"
  />
  <div
    className="loader_block loader_block--3"
  />
  <div
    className="loader_block loader_block--4"
  />
</section>
`;
